### ISSUE TITLE
one time log in

### ISSUE THREAD

**Issue by devils-rule** [PCGeekBrain/chaverim-update#11]
Is there anyway to have the app remember the last log in info, so you dont have to log in every time you open it?

**Comment by M2168J** [comment]
I'm using chrome without issue. Are u using Firefox maybe?

**Comment by PCGeekBrain** [comment]
The app will remember your login for 24 hrs after the last use.

It uses LocalStorage to store the authentication key so it will not persist in incognito mode or older browsers that do not support LocalStorage. See LocalStorage Browser Support here: https://caniuse.com/#search=LocalStorage

Which browser are you using and are you not in incognito mode?

**Comment by devils-rule** [comment]
Ok I'm using firefox in regular mode as a standalone pwa app. It has been more then 24hours since I last opened it, that could be the issue.

**Comment by PCGeekBrain** [comment]
Yes, if it was longer then 24hrs ago then you would be logged out for security.

@M2168J Should we increase this timeframe?

**Comment by devils-rule** [comment]
Can we do 48 hours, since you have shabbos and slow days. We don't want to inconvenience active members. 

@PCGeekBrain also will the notifications still come through even after the auto logout?

**Comment by M2168J** [comment]
I personally don't think we should have auto log out. Let me ask our
coordinator. Joe and Mendel, hang on

**Comment by PCGeekBrain** [comment]
@devils-rule We can definitely do 48hrs. We just need to change this config file to 2.days: https://github.com/PCGeekBrain/chaverim-update/blob/master/api/config/initializers/knock.rb

@devils-rule Yes notifications are linked to the browser via an endpoint set by the vendor and submitted to our servers. I store them in the DBS with a user_id and a tag that looks like this: "<browser> <browser version> on <platform>". E.g. "Chrome 52.0.0 on Android 5.0". I hope to have the time to add a form where the user can control which devices get notifications and allow admins to unsubscribe any users. They should even work when the page is closed as long as the browser is open. In the future (long long time from now) I hope to use WebSockets to send the call data more efficiently over the wire. In that case notifications would probably require a running app.

@M2168J Auto logout is a requirement of the technology used to secure the connection and in the future allow for social logins. However the time-frame until when the key expires is arbitrary (set by me)

**Comment by PCGeekBrain** [comment]
@M2168J have we decided what the timeframe should be?
